pyutilib/misc/timing.py

```diff
@@ -160,3 +160,305 @@ def start(self):
+        name_formatter = '{name:<' + str(max_name_len) + '}'
+        s = (name_formatter +
+             '{total_time:>15}'
+             '{num_calls:>15}'
+             '{time_per_call:>15}\n').format(name='Identifier',
+                                             total_time='Time (s)',
+                                             num_calls='# Calls',
+                                             time_per_call='Per Call (s)')
+        s += '-' * (max_name_len + 45) + '\n'
+        for name, timer in self.timers.items():
+            s += name_formatter.format(name=name)
+            s += '{0:>15.2e}'.format(timer.total_time)
+            s += '{0:>15d}'.format(timer.n_calls)
+            s += '{0:>15.2e}\n'.format(timer.total_time/timer.n_calls)
+            s += timer.pprint(indent='    ')
+        return s
+
+    def reset(self):
+        """
+        Completely reset the timer.
+        """
+        self.stack = list()
+        self.timers = dict()
+
+    def _get_timer_from_stack(self, stack):
+        """
+        This method gets the timer associated with stack.
+
+        Parameters
+        ----------
+        stack: list of str
+            A list of identifiers. 
+        
+        Returns
+        -------
+        timer: _HierarchicalHelper
+        """
+        tmp = self
+        for i in stack:
+            tmp = tmp.timers[i]
+        return tmp
+
+    def get_total_time(self, identifier):
+        """
+        Parameters
+        ----------
+        identifier: str
+            The full name of the timer including parent timers separated with dots.
+
+        Returns
+        -------
+        total_time: float
+            The total time spent with the specified timer active.
+        """
+        stack = identifier.split('.')
+        timer = self._get_timer_from_stack(stack)
+        return timer.total_time
+
+    def get_num_calls(self, identifier):
+        """
+        Parameters
+        ----------
+        identifier: str
+            The full name of the timer including parent timers separated with dots.
+
+        Returns
+        -------
+        num_calss: int
+            The number of times start_increment was called for the specified timer.
+        """
+        stack = identifier.split('.')
+        timer = self._get_timer_from_stack(stack)
+        return timer.n_calls
+
+    def get_relative_percent_time(self, identifier):
+        """
+        Parameters
+        ----------
+        identifier: str
+            The full name of the timer including parent timers separated with dots.
+
+        Returns
+        -------
+        percent_time: float
+            The percent of time spent in the specified timer 
+            relative to the timer's immediate parent.
+        """
+        stack = identifier.split('.')
+        timer = self._get_timer_from_stack(stack)
+        parent = self._get_timer_from_stack(stack[:-1])
+        return timer.total_time / parent.total_time * 100
```

Comment:
For timing very short times on machines with low precision, `total_time` can be 0

Reply:
I fixed this in the printing methods. However, I think a division by 0 error is appropriate here. Other methods are available for users to easily get the numerator and denominator individually. Although, given what I just said, it might make sense to remove the percent methods. Thoughts?

Reply:
Personally, I think that either `get_relative_percent` should return `0` when `total_time` is 0 so that clients that are doing the equivalent of `'%d' % (t.get_relative_percent)` will not get exceptions.  It is numerically incorrect, but doesn't meaningfully change the interpretation of the results.

Reply:
Another edge case: if you call `get_relative_percent` from the top of the stack, you should get `100`.

Reply:
Good catch. That should now be fixed.

Reply:
I disagree - I think returning `0` from `get_relative_percent_time` when `parent.total_time` is `0` does change the interpretation of the results. I would rather the user get an exception. However, I am happy to remove this method completely and let users do what they want manually (with calls to `get_total_time`).

Reply:
Actually, we could return `math.nan`, which is accurate and does not raise an exception. Thoughts?

Reply:
@michaelbynum Great idea!  Just make it `float('nan')` (for compatibility).